tests/basic/pure_test.go

```diff
@@ -4143,3 +4143,129 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
 	})
 
 })
+
+var _ = Describe("{TestParallelPxAndFadaVolumeResize}", func() {
+	/*
+		https://purestorage.atlassian.net/browse/PTX-23985
+		1. Deploy an app which uses FADA volume( uses csi provisioner and a pure_block backend) and an app which uses base volume(uses portworx-volume as provisioner)
+		2. Parallely resize the pvc of FADA deployed app and also resize the base app pvc
+		3. Validate the volume resize
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("TestParallelPxAndFadaVolumeResize", "Px Volume Resize in parallel to FADA/FBDA Volume Resize ( PVC Resize )", nil, 0)
+	})
+	var contexts []*scheduler.Context
+	itLog := "Px Volume Resize in parallel to FADA/FBDA Volume Resize ( PVC Resize )"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		var resizedVols []*volume.Volume
+
+		stepLog := "Deploy applications"
+		Step(stepLog, func() {
+
+			Inst().AppList = []string{"fio-cloudsnap"}
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplications(fmt.Sprintf("baseapps-%d", i))...)
+			}
+
+		})
+
+		stepLog = "Deploy a app which uses FADA volume and validate"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			context, err := Inst().S.Schedule("test-fada-volume", scheduler.ScheduleOptions{
+				AppKeys:            []string{"fio-fa-davol"},
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+				PvcSize:            6 * units.GiB,
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", "test-fada-volume")
+			contexts = append(contexts, context...)
+
+		})
+		ValidateApplications(contexts)
+
+		stepLog = "Do parallel resize of base apps volume and FADA volume"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+
+			go func() {
+				for _, ctx := range contexts[:1] {
+					var appVolumes []*volume.Volume
+					var err error
+					var pvcs []*v1.PersistentVolumeClaim
+					appVolumes, err = Inst().S.GetVolumes(ctx)
+					log.FailOnError(err, "Failed to get volumes from context")
+					log.InfoD(fmt.Sprintf("increase volume size %s on app %s's volumes: %v",
+						Inst().V.String(), ctx.App.Key, appVolumes))
+					for _, eachVol := range appVolumes {
+						pvc, err := GetPVCObjFromVol(eachVol)
+						log.FailOnError(err, "Failed to get PVC Details from Volume [%v]", eachVol.Name)
+						pvcs = append(pvcs, pvc)
+
+					}
+					for _, pvc := range pvcs {
+						pvcSize := pvc.Spec.Resources.Requests.Storage().String()
```

Comment:
pls use Size() , using Size() , there is no need to do trimming

Reply:
i have compared total of three size(),v1.ResourceStorage, and the above one which i implemented, among all three the one which have given right results so used that one